src/librustc/middle/traits/select.rs

```diff
@@ -1045,6 +1049,11 @@ impl<'cx, 'tcx> SelectionContext<'cx, 'tcx> {
                 }
             });
         }
+
+        if self.tcx().default_trait_impls.borrow().contains(&def_id) {
```

Comment:
I think we should probably wrap this in some more attractive accessor (`ty::trait_has_default_impl()`)